File: src/main/kotlin/driver/PageProvider.java
package driver;

import com.microsoft.playwright.*;
import config.Config;

public final class PageProvider {
    private PageProvider() {}

    public static BrowserContext newContext(Browser browser) {
        Browser.NewContextOptions ctxOpts = new Browser.NewContextOptions();
        BrowserContext ctx = browser.newContext(ctxOpts);
        ctx.setDefaultTimeout(Config.timeoutMs());
        ctx.setDefaultNavigationTimeout(Config.timeoutMs());
        return ctx;
    }

    public static Page newPage(BrowserContext context) {
        return context.newPage();
    }
}

File: src/main/kotlin/driver/PageProvider.kt
package driver

import com.microsoft.playwright.Browser
import com.microsoft.playwright.BrowserContext
import com.microsoft.playwright.Page
import config.Config

object PageProvider {
    fun newContext(browser: Browser): BrowserContext {
        val ctxOpts = Browser.NewContextOptions()
        val ctx = browser.newContext(ctxOpts)
        ctx.setDefaultTimeout(Config.timeoutMs().toDouble())
        ctx.setDefaultNavigationTimeout(Config.timeoutMs().toDouble())
        return ctx
    }

    fun newPage(context: BrowserContext): Page = context.newPage()
}

